src/backend/query_builder.rs

```diff
@@ -1177,6 +1198,70 @@ pub trait QueryBuilder: QuotedBuilder {
         }
     }
 
+    #[doc(hidden)]
+    /// Translate [`Frame`] into SQL statement.
+    fn prepare_frame(&self, frame: &Frame, sql: &mut SqlWriter, collector: &mut dyn FnMut(Value)) {
+        match *frame {
+            Frame::UboundedPreceding => write!(sql, " UNBOUNDED PRECEDING ").unwrap(),
+            Frame::Preceding(v) => {
+                self.prepare_value(&Some(v).into(), sql, collector);
+                write!(sql, " PRECEDING ").unwrap();
+            }
+            Frame::CurrentRow => write!(sql, " CURRENT ROW ").unwrap(),
+            Frame::Following(v) => {
+                self.prepare_value(&Some(v).into(), sql, collector);
+                write!(sql, " FOLLOWING ").unwrap();
+            }
+            Frame::UnboundedFollowing => write!(sql, " UNBOUNDED FOLLOWING ").unwrap(),
+        }
+    }
+
+    #[doc(hidden)]
+    /// Translate [`WindowStatement`] into SQL statement.
+    fn prepare_window_statement(
+        &self,
+        window: &WindowStatement,
+        sql: &mut SqlWriter,
+        collector: &mut dyn FnMut(Value),
+    ) {
+        let (first, exprs) = &window.partition_by;
+        write!(sql, " PARTITION BY ").unwrap();
+        self.prepare_simple_expr(first, sql, collector);
+
+        if !exprs.is_empty() {
+            for expr in exprs {
+                write!(sql, ", ").unwrap();
+                self.prepare_simple_expr(expr, sql, collector);
+            }
+        }
+
+        if !window.order_by.is_empty() {
+            write!(sql, " ORDER BY ").unwrap();
+            window.order_by.iter().fold(true, |first, expr| {
+                if !first {
+                    write!(sql, ", ").unwrap()
+                }
+                self.prepare_order_expr(expr, sql, collector);
+                false
+            });
+        }
+
+        if let Some(frame) = &window.frame {
+            match frame.r#type {
+                FrameType::Range => write!(sql, " RANGE ").unwrap(),
+                FrameType::Rows => write!(sql, " ROWS ").unwrap(),
+            };
+            if let Some(end) = &frame.end {
+                write!(sql, " BETWEEN ").unwrap();
```

Comment:
```suggestion
                write!(sql, "BETWEEN ").unwrap();
```

I guess we have an excessive space before `BETWEEN`?